doc/tutorials/platformer/app/lib/actors/ember.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:ember_quest/ember_quest.dart';
+import 'package:ember_quest/objects/ground_block.dart';
+import 'package:ember_quest/objects/platform_block.dart';
+import 'package:ember_quest/objects/star.dart';
+import 'package:flame/collisions.dart';
+import 'package:flame/components.dart';
+import 'package:flame/effects.dart';
+import 'package:flutter/services.dart';
+
+class EmberPlayer extends SpriteAnimationComponent
+    with KeyboardHandler, CollisionCallbacks, HasGameRef<EmberQuestGame> {
+  EmberPlayer({
+    required super.position,
+  }) : super(size: Vector2.all(64), anchor: Anchor.center);
+
+  final Vector2 velocity = Vector2.zero();
+  final Vector2 fromAbove = Vector2(0, -1);
+  final double gravity = 15;
+  final double jumpSpeed = 600;
+  final double moveSpeed = 200;
+  int horizontalDirection = 0;
+
+  bool hasJumped = false;
+  bool isOnGround = false;
+  bool hitByEnemy = false;
+
+  @override
+  Future<void> onLoad() async {
+    animation = SpriteAnimation.fromFrameData(
+      gameRef.images.fromCache('ember.png'),
+      SpriteAnimationData.sequenced(
+        amount: 4,
+        textureSize: Vector2.all(16),
+        stepTime: 0.12,
+      ),
+    );
+
+    add(
+      CircleHitbox()..collisionType = CollisionType.active,
+    );
+  }
+
+  @override
+  bool onKeyEvent(RawKeyEvent event, Set<LogicalKeyboardKey> keysPressed) {
+    horizontalDirection = 0;
+    horizontalDirection += (keysPressed.contains(LogicalKeyboardKey.keyA) ||
+            keysPressed.contains(LogicalKeyboardKey.arrowLeft))
+        ? -1
+        : 0;
+    horizontalDirection += (keysPressed.contains(LogicalKeyboardKey.keyD) ||
+            keysPressed.contains(LogicalKeyboardKey.arrowRight))
+        ? 1
+        : 0;
+
+    hasJumped = keysPressed.contains(LogicalKeyboardKey.space);
+    return true;
+  }
+
+  @override
+  void update(double dt) {
+    velocity.x = horizontalDirection * moveSpeed;
+    gameRef.objectSpeed = 0;
+    //Prevent ember from going backwards at screen edge
+    if (position.x - 36 <= 0 && horizontalDirection < 0) {
+      velocity.x = 0;
+    }
+    //Prevent ember from going beyond half screen
+    if (position.x + 64 >= gameRef.size.x / 2 && horizontalDirection > 0) {
+      velocity.x = 0;
+      gameRef.objectSpeed = -moveSpeed;
+    }
+
+    //Apply basic gravity
+    velocity.y += gravity;
+
+    //Determine if ember has jumped
+    if (hasJumped) {
+      if (isOnGround) {
+        velocity.y = -jumpSpeed;
+        isOnGround = false;
+      }
+      hasJumped = false;
+    }
+
+    //Prevent ember from jumping to crazy fast
+    velocity.y = velocity.y.clamp(-jumpSpeed, 150);
+
+    //Adjust ember position
+    position += velocity * dt;
+
+    //If ember fell in pit, then game over
+    if (position.y > gameRef.size.y + 64) {
+      gameRef.health = 0;
+    }
+
+    if (gameRef.health <= 0) {
+      removeFromParent();
+    }
+
+    // Flip ember if needed.
```

Comment:
Nice with a full stop here, maybe use that on all of the comments?

Reply:
Fixed.